singularity-eos/eos/eos_variant.hpp

```diff
@@ -980,12 +1001,18 @@ class Variant {
 
   inline constexpr Variant UnmodifyOnce() {
     return mpark::visit(
-        [](auto &eos) { return eos_variant<EOSs...>(eos.UnmodifyOnce()); }, eos_);
+        [](auto &eos) -> eos_variant<EOSs...> {
```

Comment:
were there issues with implicit conversions?

Reply:
Uh I don't remember why I had to make this change---I think there was an error about visitor always returning the same type.